src/transactions/simulation/TxSimUtils.cpp

```diff
@@ -428,10 +522,87 @@ mutateScaledOperation(Operation& op, uint32_t partition)
             generateScaledClaimableBalanceID(
                 op.body.claimClaimableBalanceOp().balanceID.v0(), partition);
         break;
+    case BEGIN_SPONSORING_FUTURE_RESERVES:
+        mutateScaledAccountID(
+            op.body.beginSponsoringFutureReservesOp().sponsoredID, partition);
+        break;
+    case REVOKE_SPONSORSHIP:
+        if (op.body.revokeSponsorshipOp().type() ==
+            REVOKE_SPONSORSHIP_LEDGER_ENTRY)
+        {
+            auto& key = op.body.revokeSponsorshipOp().ledgerKey();
+            switch (key.type())
+            {
+            case ACCOUNT:
```

Comment:
The logic here is identical to `TxSimUtils::generateScaledDeadEntries`, so we should refactor this to avoid duplication

Reply:
Done.